pandas/core/index.py

```diff
@@ -3170,6 +3235,18 @@ def __array__(self, dtype=None):
         """ the array interface, return my values """
         return np.array(self._data, dtype=dtype)
 
+    @cache_readonly
+    def _isnan(self):
+        """ return if each value is nan"""
+        return self._data.codes == -1
+
+    @Appender(_index_shared_docs['fillna'])
+    def fillna(self, value, downcast=None):
+        if not lib.isscalar(value):
```

Comment:
this looks good here. I agree more appropriate to defer to the actual object here.